src/lib.rs

```diff
@@ -2950,6 +2951,117 @@ fn digraph_transitivity(graph: &digraph::PyDiGraph) -> f64 {
     }
 }
 
+pub fn _core_number<Ty>(
+    py: Python,
+    graph: &StableGraph<PyObject, PyObject, Ty>,
+) -> PyResult<PyObject>
+where
+    Ty: EdgeType,
+{
+    let node_num = graph.node_count();
+    let mut cores: HashMap<NodeIndex, usize> = HashMap::with_capacity(node_num);
+    let mut node_vec: Vec<NodeIndex> = graph.node_indices().collect();
+    let mut degree_map: HashMap<NodeIndex, usize> =
+        HashMap::with_capacity(node_num);
+    let mut nbrs: HashMap<NodeIndex, Vec<NodeIndex>> =
+        HashMap::with_capacity(node_num);
+    let mut node_pos: HashMap<NodeIndex, usize> =
+        HashMap::with_capacity(node_num);
+
+    for k in node_vec.iter() {
+        let k_nbrs: Vec<NodeIndex> = graph.neighbors_undirected(*k).collect();
+        let k_deg = k_nbrs.len();
+
+        nbrs.insert(*k, k_nbrs);
+        cores.insert(*k, k_deg);
+        degree_map.insert(*k, k_deg);
+    }
+    node_vec.par_sort_by_key(|k| degree_map.get(k));
+
+    let mut bin_boundaries = vec![0];
```

Comment:
Is there a better estimate we can come up with for the size of this Vec? Right now this will end up with a bunch of reallocations because every time we push onto into we'll have to grow (since it's initial allocation will be for 1 element). Not critical just me idly wondering if there is a possible optimization here.

Reply:
That's a good point, the size of `bin_boundaries` is precisely `max_degree + 1`. I'll push a quick fix